squire_sdk/src/client/tournaments.rs

```diff
@@ -239,7 +259,7 @@ impl ManagerState {
                     let (sink, stream) = ws.split();
                     let (broad, sub) = watch_channel(());
                     entry.get_mut().comm = Some((sink, broad));
-                    scheduler.add_stream(stream);
+                    scheduler.attach_stream(stream.fuse());
```

Comment:
Seems like `.fuse()` showed up during the rebase. I tried removing it, and I get a type error in the front end code. I guess this is something I would have seen if I had compiled the front end before the rebase.

But the weird thing is that `.fuse()` is also called within `troupe`, in `attach_stream_inner`. Is this correct, and is it necessary?

Reply:
This is really more of an issue with `troupe`. I remember writing this function signature and was torn on this question. `fuse` adds (basically) no overhead, so it isn't a problem. You need to give `attached_stream` a fused stream because I want the caller to acknowledge that the stream is capable of terminating.

Internally, the second call to `fuse` can be gotten rid of. If you would like, you can open a PR on the troupe repo to address this.